thinc/loss.py

```diff
@@ -126,34 +147,63 @@ def convert_truths(self, truths, guesses: Floats2d) -> Tuple[Floats2d, Floats2d]
     def __call__(
         self, guesses: Floats2d, truths: IntsOrFloatsOrStrs
     ) -> Tuple[Floats2d, float]:
-        d_truth = self.get_grad(guesses, truths)
-        return (d_truth, self._get_loss_from_grad(d_truth))
-
-    def get_grad(self, guesses: Floats2d, truths: IntsOrFloatsOrStrs) -> Floats2d:
         target, mask = self.convert_truths(truths, guesses)
+        self._validate_input(guesses, target)
+        d_truth = self._get_grad(guesses, target, mask)
+        loss = self._get_loss(guesses, target, mask)
+        return (d_truth, loss)
+
+    def _validate_input(self, guesses: Floats2d, target: Floats2d) -> None:
         xp = get_array_module(target)
+        if not xp.allclose(guesses.sum(axis=1), 1.0):
```

Comment:
I think it's good to do input validation, but do you have an idea of the impact in CPU and GPU training?

Reply:
Not sure honestly yet, but I think this is really important, because if the rows of `truths` and `guesses` don't represent categorical distributions the loss value doesn't make sense.

Reply:
What do you think?

Reply:
Measure :).

Reply:
I think this is addressed by now right @danieldk ? in the new `_validate_input` of the `CategoricalCrossentropyBase` ?